abcclassroom/tests/test_git.py

```diff
@@ -1,28 +1,206 @@
+    without needing to create files in our tests each time."""
+
+    repo_dir = Path(tmpdir, "assignment-1", "course-test-student")
+    repo_dir.mkdir(parents=True, exist_ok=True)
+    a_file = Path(repo_dir, "README.md")
+    a_file.write_text("Some text in the readme")
+
+    return repo_dir
+
+
+# TODO: add an  actual check here for the standard out
+def test_check_git_ssh_pass():
+    """When ssh is setup correctly, the check should run and pass with no
+    output."""
+    with mock.patch("subprocess.run"):
+        # Skip actually running the subprocess call and return the expected
+        # output
+        subprocess.run.side_effect = subprocess.CalledProcessError(
+            returncode=1, cmd="", stderr="Hi username"
+        )
+        # Here this should just run and pass
+        # I don't think we need an assert unless we add a message for
+        # successful check to the function. Then we get a return output that
+        # can be tested
+        abcgit.check_git_ssh()
+
+
+def test_check_git_ssh_warning(capsys):
+    """Test what happens when ssh is setup but user hasn't logged in"""
+    with mock.patch("subprocess.run"):
+        # Skip actually running the subprocess call and return the expected
+        # output. We always trigger a CalledProcessError in this function
+        subprocess.run.side_effect = subprocess.CalledProcessError(
+            returncode=1, cmd="", stderr="Warning: Permanently"
+        )
+
+        abcgit.check_git_ssh()
+        captured_output = capsys.readouterr().out.splitlines()
+        assert captured_output[0].startswith("Warning:")
+
+
+def test_check_git_ssh_error(capsys):
+    """Test what happens when ssh is not setup - should raise RuntimeError"""
+    with mock.patch("subprocess.run"):
+        # Skip running subprocess call, return the expected error
+        subprocess.run.side_effect = subprocess.CalledProcessError(
+            returncode=1, cmd="", stderr="Encountered this error"
+        )
+        with pytest.raises(RuntimeError):
+            abcgit.check_git_ssh()
+        captured_output = capsys.readouterr().out.splitlines()
+        assert captured_output[0].startswith("Encountered this error")
+
+
+# NOTE: Updated to run git status so it works universally
+# TODO: i am cheating and mocking git status and forcing a return. is this
+#  even useful? Also does status really need to be mocked? Does CI have status
+# setup because perhaps this is something we can just run without a mock?
+def test_call_git_status(capsys):
+    """Test that _call_git helper function runs as expected"""
+    with mock.patch("abcclassroom.git._call_git", return_value="On branch"):
+        ret = abcgit._call_git("status")
+    assert "On branch" in ret
+
+
+# TODO i am not sure how to set this up but what happens if git isn't setup
+#  and you try to run _call_git?
+
+
+def test_clone_repo_pass(monkeypatch, example_student_repo, capsys, tmp_path):
```

Comment:
this could run IF we could figure out authentication. i really do not understand how to do that part.